src/chat/bot.py

```diff
@@ -0,0 +1,78 @@
+from typing import final
+
+from google.genai.types import HarmBlockThreshold, HarmCategory
+from pydantic_ai import Agent
+from pydantic_ai.messages import UserPromptPart
+from pydantic_ai.models.google import GoogleModel, GoogleModelSettings
+from pydantic_ai.providers.google import GoogleProvider
+from pydantic_ai.toolsets import AbstractToolset
+
+from chat.memory import retrieve_conversation, set_conversation
+from chat.memory import retrieve_conversation, set_conversation
+from chat.types import Dependencies, UserId
+from chat.types import Dependencies, UserId
+from env import Environment
+
+
+@final
+class Bot:
+    def __init__(
+        self,
+        *,
+        env: Environment,
+        toolset: AbstractToolset[Dependencies],
+        chats: dict[UserId, Agent[Dependencies, str]] | None = None,
+    ) -> None:
+        self.chats = chats or {}
+        self.__env = env
+        self.__toolset = toolset
+
+    def make_agent(self) -> Agent[Dependencies, str]:
+        return Agent(
+            GoogleModel(
+                "gemini-2.5-pro",
+                provider=GoogleProvider(api_key=self.__env.google_cloud_api_key),
+                settings=GoogleModelSettings(
+                    temperature=0.2,
```

Comment:
```suggestion
                    temperature=0.6,
```
We tested the system prompt for this temperature and it worked best for multi-language responses

Reply:
Ok